packages/atlas/src/components/_auth/ExternalSignInModal/ExternalSignInSteps/ExternalSignInModalWalletStep/ExternalSignInModalWalletStep.tsx

```diff
@@ -87,14 +96,27 @@ export const ExternalSignInModalWalletStep: FC<ExternalSignInModalWalletStepProp
   const selectedWallet = (selectedWalletIdx != null && wallets[selectedWalletIdx]) || null
 
   const handleConfirm = useCallback(async () => {
-    if (!selectedWallet) return
+    if (selectedWalletIdx < wallets.length && !selectedWallet) return
 
     setIsConnecting(true)
     setHasError(false)
-    const accounts = await signInToWallet(selectedWallet.extensionName)
+
+    const accounts =
+      selectedWallet?.extensionName === 'WalletConnect'
```

Comment:
We're doing this check with hardcoded name in a couple of places. I think we can use the utility `isWalletConnectWallet` you've created instead?